pkg/datapath/linux/config/config.go

```diff
@@ -454,24 +454,14 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 	}
 
 	if iface := option.Config.EncryptInterface; len(iface) != 0 {
-		// When FIB lookup is not supported (older kernels)  we need to
-		// pick an interface so pick first interface in list. Then we pick
-		// an IPv4 address to use by selecting link IPAddr. In case with
-		// kernel support, the kernel datapath will use the FIB lookup helper
-		// and this define is ignored.
-		link, err := netlink.LinkByName(iface[0])
-		if err == nil {
-			cDefinesMap["ENCRYPT_IFACE"] = fmt.Sprintf("%d", link.Attrs().Index)
-
-			addr, err := netlink.AddrList(link, netlink.FAMILY_V4)
-			if err != nil {
-				return err
-			}
-			if len(addr) == 0 {
-				return fmt.Errorf("no IPv4 addresses available in encrypt interface %q", iface)
+		a := byteorder.HostSliceToNetwork(node.GetIPv4(), reflect.Uint32).(uint32)
+		cDefinesMap["IPV4_ENCRYPT_IFACE"] = fmt.Sprintf("%d", a)
```

Comment:
nit, I just noticed that `IPV4_ENCRYPT_IFACE` isn't an interface, it's an address :open_mouth: . I guess it's always been like this but a little surprising (especially since the similarly-named `ENCRYPT_IFACE` is an ifindex for a network interface).

Reply:
oh, I guess I see the relationship here, `IPV4_ENCRYPT_IFACE` was the IPv4 address associated with `ENCRYPT_IFACE`. Except now it's not, it's the cilium_host IP.

Reply:
Agree bad choice of names but renaming it now seems likely to just cause more confusion during backports.